jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPart.java

```diff
@@ -631,20 +631,20 @@ public Content.Chunk read()
                     Content.Chunk chunk = part.getContent().read();
                     if (chunk == null || chunk instanceof Content.Chunk.Error)
                         yield chunk;
-                    if (chunk.isLast())
+                    if (!chunk.isLast())
+                        yield chunk;
+                    state = State.MIDDLE;
+                    if (chunk.hasRemaining())
                     {
-                        if (!chunk.hasRemaining())
-                        {
-                            chunk.release();
-                            chunk = Content.Chunk.EMPTY;
-                        }
-                        else
-                        {
-                            chunk = Content.Chunk.from(chunk.getByteBuffer(), false, chunk);
-                        }
-                        state = State.MIDDLE;
+                        Content.Chunk wrapped = Content.Chunk.from(chunk.getByteBuffer(), false, chunk);
+                        chunk.release();
```

Comment:
It's kind of annoying that we know `Chunk.from` is going to do a retain, that counter acts this release.    Perhaps would be nice to have a version of `from` that doesn't do the retain, so we don't have to do the release?

Reply:
I now see that a lot of places the implicit `retain()` that was added has to be compensated by a `release()` and @gregw is proposing a version of `Chunk.from` that doesn't `retain()` to avoid two CAS operations and help readability.

`Chunk.from` used to not retain, why was this changed?